### PR TITLE
Remove middleware - part 1

### PR DESCRIPTION
**Pull request by jdisanti**
This PR removes the runtime mode flag, test compile flags, and CI checks for client middleware.

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/IntegrationTestDependencies.kt [hunk 1]
```diff
@@ -87,21 +88,18 @@ class IntegrationTestDependencies(
...
                 val smithyTypes = CargoDependency.smithyTypes(codegenContext.runtimeConfig)
                     .copy(features = setOf("test-util"), scope = DependencyScope.Dev)
+                addDependency(awsRuntime(runtimeConfig).toDevDependency().withFeature("test-util"))
+                addDependency(FuturesUtil)
+                addDependency(SerdeJson)
                 addDependency(smithyAsync)
                 addDependency(smithyClient)
+                addDependency(smithyProtocolTestHelpers(codegenContext.runtimeConfig))
+                addDependency(smithyRuntime(runtimeConfig).copy(features = setOf("test-util"), scope = DependencyScope.Dev))
+                addDependency(smithyRuntimeApi(runtimeConfig).copy(features = setOf("test-util"), scope = DependencyScope.Dev))
```
[the remark this answers is not in the document]
Comment (Velfi) [reply aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/IntegrationTestDependencies.kt:98]: Also, what do you think of adding a method to `CargoDependency` to convert a dep into a dev dependency. That way we'd write `smithyRuntime(runtimeConfig).devDependency()` or something.

File: aws/sdk/integration-tests/kms/tests/retryable_errors.rs [hunk 2]
```diff
@@ -3,144 +3,78 @@
...
-        let err = op.response_handler.parse(&http_response).map_err(|e| {
-            SdkError::service_error(
-                e,
-                operation::Response::new(http_response.map(SdkBody::from)),
-            )
-        });
-        let retry_kind = op.retry_classifier.classify_retry(err.as_ref());
-        assert_eq!(retry_kind, RetryKind::Error(ErrorKind::ThrottlingError));
-    }
+    dbg!(&err);
```
Comment (Velfi) [inline aws/sdk/integration-tests/kms/tests/retryable_errors.rs:48]: leftover?

File: aws/sdk/integration-tests/s3/tests/interceptors.rs [hunk 3]
```diff
@@ -3,123 +3,120 @@
...
-                .customize()
-                .await
-                .unwrap()
-                .interceptor(TestInterceptor("value2"))
-                .send()
-                .await
-        )
-        .expect_err("no fake response set");
+    // The second `TestInterceptor` will replace `value1` with `value2` in config
+    dbg!(
```
Comment (Velfi) [inline aws/sdk/integration-tests/s3/tests/interceptors.rs:72]: Is this dbg necessary for the test? Can it be removed?

File: codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/client/FluentClientGenerator.kt [hunk 4]
```diff
@@ -132,121 +127,73 @@ class FluentClientGenerator(
...
-                    ##[derive(#{Clone}, ::std::fmt::Debug)]
-                    pub struct Client {
-                        handle: #{Arc}<Handle>,
+                    ##[doc(hidden)]
+                    // TODO(enableNewSmithyRuntimeCleanup): Delete this function when cleaning up middleware
+                    // This is currently kept around so the tests still compile in both modes
+                    /// Creates a client with the given service configuration.
+                    pub fn with_config<C, M, R>(_client: #{client}::Client<C, M, R>, conf: crate::Config) -> Self {
+                        Self::from_conf(conf)
                     }
```
Comment (Velfi) [inline codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/client/FluentClientGenerator.kt:183]: We still need this for some tests, right? Same for the `conf` method below?

File: aws/sdk/integration-tests/kms/tests/retryable_errors.rs (hunk 2 above)
Comment (jdisanti) [reply aws/sdk/integration-tests/kms/tests/retryable_errors.rs:48]: I don't see this in the diff? Be sure to hide whitespace changes.

File: codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/client/FluentClientGenerator.kt (hunk 4 above)
Comment (jdisanti) [reply codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/client/FluentClientGenerator.kt:183]: Will address the TODO in a later PR. This method isn't needed anymore.

File: codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/client/FluentClientGenerator.kt [hunk 5]
```diff
@@ -132,121 +127,73 @@ class FluentClientGenerator(
...
-                            Self { handle: self.handle.clone() }
+                impl Client {
+                    /// Creates a new client from the service [`Config`](crate::Config).
+                    ///
+                    /// ## Panics
+                    ///
+                    /// - This method will panic if the `conf` is missing an async sleep implementation. If you experience this panic, set
+                    ///     the `sleep_impl` on the Config passed into this function to fix it.
+                    /// - This method will panic if the `conf` is missing an HTTP connector. If you experience this panic, set the
+                    ///     `http_connector` on the Config passed into this function to fix it.
```
[the remark this answers is not in the document]
Comment (jdisanti) [reply codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/client/FluentClientGenerator.kt:152]: Not accurate. Fixed in https://github.com/awslabs/smithy-rs/pull/2923.

File: aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/IntegrationTestDependencies.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (jdisanti) [reply aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/IntegrationTestDependencies.kt:98]: It's a good idea, but I think out of scope for this PR.

**Pull request merged**